workos/resources/list.py

```diff
@@ -117,56 +124,55 @@ def auto_paging_iter(self):
     DirectoryGroup,
     DirectoryUser,
 )
+FilterParams = TypeVar("FilterParams")
 
 
 class ListMetadata(BaseModel):
     after: Optional[str] = None
     before: Optional[str] = None
 
 
-class ListPage(BaseModel, Generic[ListableResource]):
+class ListPage(WorkOSModel, Generic[ListableResource]):
     object: Literal["list"]
     data: List[ListableResource]
     list_metadata: ListMetadata
 
 
-class ListArgs(BaseModel, extra="allow"):
-    limit: Optional[int] = 10
-    before: Optional[str] = None
-    after: Optional[str] = None
-    order: Literal["asc", "desc"] = "desc"
-
-    class Config:
-        extra = "allow"
+class ListArgs(TypedDict):
+    limit: int
+    before: Optional[str]
+    after: Optional[str]
+    order: Literal["asc", "desc"]
 
 
-class WorkOsListResource(BaseModel, Generic[ListableResource]):
+class WorkOsListResource(
+    WorkOSModel,
+    Generic[ListableResource, FilterParams],
+):
     object: Literal["list"]
     data: List[ListableResource]
     list_metadata: ListMetadata
 
-    # These fields end up exposed in the types. Does we care?
     list_method: Callable = Field(exclude=True)
     list_args: ListArgs = Field(exclude=True)
+    filter_params: FilterParams = Field(exclude=True)
```

Comment:
I don't love passing in an entirely separate type for this, but guess what? Python type hinting doesn't support intersection between two `TypedDict`s. This is why we can't have nice things.

https://github.com/python/typing/discussions/1275

Reply:
... actually, I wonder if all of the list resource params can inherit from `ListArgs` and add their own specific params. I think that will work out better. Going to convert this back to drafe.